Playmate/View Controllers/Home Tab/HomeViewController.m

```diff
@@ -26,187 +26,141 @@ @interface HomeViewController () <UITableViewDelegate, UITableViewDataSource, DZ
+
+	[query orderByAscending:@"occursAt"];
+
+	// fetch data asynchronously
+	[query findObjectsInBackgroundWithBlock:^(NSArray *sessions, NSError *error) {
+	         if (sessions != nil) {
+
+			 [self.sessionList removeAllObjects];
+			 [self filterSessions:sessions];
+
+			 [self.tableView reloadData];
+		 } else {
+			 NSLog(@"%@", error.localizedDescription);
+		 }
+	         [self.refreshControl endRefreshing];
+	 }];
 }
 
 // Filter out sessions that do not contain self
 - (void)filterSessions:(NSArray *)sessions {
-    
-    NSMutableArray *tempList = (NSMutableArray *)sessions;
-    PFUser *currUser = [[PFUser currentUser] fetchIfNeeded];
-    
-    for (Session *sesh in tempList) {
-        
-        for (PFUser *user in sesh[@"playersList"]) {
-            [user fetchIfNeeded];
-            
-            NSDate *now = [NSDate date];
-            NSComparisonResult result = [now compare:sesh.occursAt];
-            
-            if ([currUser.username isEqualToString:user.username] && result == NSOrderedAscending) {
-                [self.sessionList addObject:sesh];
-                break;
-            }
-        }
-    }
-}
+	NSMutableArray *tempList = (NSMutableArray *)sessions;
+	PFUser *currUser = [[PFUser currentUser] fetchIfNeeded];
 
-#pragma mark - Empty table view protocol methods
+	for (Session *sesh in tempList) {
 
-- (UIImage *)imageForEmptyDataSet:(UIScrollView *)scrollView
-{
-    return [UIImage imageNamed:@"logo_small"];
-}
+		for (PFUser *user in sesh[@"playersList"]) {
+			[user fetchIfNeeded];
 
-- (NSAttributedString *)titleForEmptyDataSet:(UIScrollView *)scrollView
-{
-    NSString *text = [Constants emptyTablePlaceholderTitle];
-    
-    NSDictionary *attributes = @{NSFontAttributeName: [UIFont boldSystemFontOfSize:18.0f],
-                                 NSForegroundColorAttributeName: [UIColor darkGrayColor]};
-    
-    return [[NSAttributedString alloc] initWithString:text attributes:attributes];
-}
-
-- (NSAttributedString *)descriptionForEmptyDataSet:(UIScrollView *)scrollView
-{
-    NSString *text = [Constants emptyTablePlaceholderMsg];
-    
-    NSMutableParagraphStyle *paragraph = [NSMutableParagraphStyle new];
-    paragraph.lineBreakMode = NSLineBreakByWordWrapping;
-    paragraph.alignment = NSTextAlignmentCenter;
-    
-    NSDictionary *attributes = @{NSFontAttributeName: [UIFont systemFontOfSize:14.0f],
-                                 NSForegroundColorAttributeName: [UIColor lightGrayColor],
-                                 NSParagraphStyleAttributeName: paragraph};
-                                 
-    return [[NSAttributedString alloc] initWithString:text attributes:attributes];
-}
+			NSDate *now = [NSDate date];
+			NSComparisonResult result = [now compare:sesh.occursAt];
 
-- (UIColor *)backgroundColorForEmptyDataSet:(UIScrollView *)scrollView
-{
-    return [Constants playmateBlue];
+			if ([currUser.username isEqualToString:user.username] && result == NSOrderedAscending) {
```

Comment:
to search an object you can also use: `https://developer.apple.com/documentation/foundation/nsarray/1408043-indexofobjectpassingtest`